platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemBuilder.java

```diff
@@ -111,8 +111,8 @@ public InternalProblem build() {
         }
 
         Throwable exceptionForProblemInstantiation = getExceptionForProblemInstantiation();
-        if (problemStream != null) {
-            addLocationsFromProblemStream(this.locations, exceptionForProblemInstantiation);
+        if (problemStream != null && collectLocation) {
```

Comment:
🤔 It seems to me that we should always collect the stacktrace location. Though `collectLocation` decides where it ends up:
- `collectLocation == true` -> originLocations
- `collectLocation == false` -> contextLocations.